bindings/python/src/PyXRootDFileSystem.cc

```diff
@@ -608,15 +608,15 @@ namespace PyXRootD
   {
     static const char         *kwlist[] = { "files", "flags", "priority",
                                             "timeout", "callback", NULL };
-    XrdCl::PrepareFlags::Flags flags;
+    uint16_t                   flagval  = 0;
     uint8_t                    priority = 0;
     uint16_t                   timeout  = 0;
     PyObject                  *pyfiles = NULL, *callback = NULL;
     PyObject                  *pyresponse = NULL, *pystatus = NULL;
     XrdCl::XRootDStatus        status;
 
-    if ( !PyArg_ParseTupleAndKeywords( args, kwds, "OK|bHO:prepare",
-         (char**) kwlist, &pyfiles, &flags, &priority, &timeout, &callback ) )
+    if ( !PyArg_ParseTupleAndKeywords( args, kwds, "OH|bHO:prepare",
```

Comment:
This is the only line mentioned as the fix in #1810. Just out of curiosity, why are the other changes necessary?

Reply:
The difference with respect to my suggestion in #1810 was to change the storage type of the variable used to store the flags-value, that PyArg_ParseTupleAndKeywords() fills, to be at least 16bits. Then the value is static_cast to an XrdCl::PrepareFlags::Flags (an enum). ParseTupleAndKeywords() needs storage large enough to be consistent with its conversion string (which was the cause of the crash of #1810) and I thought the storage size of PrepareFlags::Flags wasn't obvious:

The storage used for the enum should be an integer type large enough to hold the enumerated values. Looking at the enum definition the largest value would be 'Evict', which needs 9 bits, so the storage is infact going to be at least 16 bits. Also both gcc and clang (that I tried) use at least 32 bits even for small enums. But I thought this was all quite some reasoning for each person that reads this source, and it would be clearer to make an explicit conversion. However there are other places at PyXRootDFileSystem.cc that use ParseTupleAndKeywords to store directly to an enum and I'm not proposing to change them with no other reason.

So if you'd suggest to change it back to the shorter version, possibly with a comment, I'd be very fine with that! (I'm not going to merge this PR, at least not before next week, that will also let Michal have a look too).

While preparing this I've seen some other issues in the python bindings (python ref count leaks & the file line iteration, which doesn't work in python3 and some things like attribute name have to be given as byte strings despite messages which say they have to be strings); so I would like to open some more PRs for those soon. While testing I've also been using the CI tests with python3, with some conversion, and porting the tests to python3 was also the suggestion of issue #1588, so perhaps I could also address that issue by changing the tests. (But best to keep all these as separate PRs I think).